gym_multigrid/envs/lavarooms.py

```diff
@@ -0,0 +1,332 @@
+                actions=self.actions_set,
+                type="agent",
+            )
+        ]
+
+        self.doorway_positions = [(10, 6), (2, 6)]
+        self.vert_wall_positions = [(6, 0), (7, 6)]
+        self.hor_wall_positions = [(0, 6), (6, 6)]
+
+        self.goal_positions = [(6, 3), (1, 1)]  # (7, 1)
+        self.agent_positions = [(6, 9), (1, 11)]
+        self.lava_positions = [
+            [
+                (2, 6),
+                (2, 7),
+                (2, 8),
+                (3, 7),
+                (3, 8),
+                (4, 7),
+                (4, 8),
+            ],
+            [
+                (10, 6),
+                (10, 7),
+                (10, 8),
+                (9, 7),
+                (9, 8),
+                (8, 7),
+                (8, 8),
+            ],
+        ]
+
+        super().__init__(
+            width=self.width,
+            height=self.height,
+            max_steps=max_steps,
+            see_through_walls=see_through_walls,
+            agents=self.agents,
+            agent_view_size=agent_view_size,
+            actions_set=self.actions_set,
+            partial_obs=partial_observability,
+            world=self.world,
+            render_mode=render_mode,
+            highlight_visible_cells=highlight_visible_cells,
+            tile_size=tile_size,
+        )
+
+    def _gen_grid(self, width, height):
+        # Create the grid
+        self.grid = Grid(width, height, self.world)
+
+        # Generate the surrounding walls
+        self.grid.horz_wall(0, 0)
+        self.grid.horz_wall(0, height - 1)
+        self.grid.vert_wall(0, 0)
+        self.grid.vert_wall(width - 1, 0)
+
+        room_w = width // 2
+        room_h = height // 2
+
+        # # Bottom wall and door
+        # for coord in vert_wall_positions:
+        #     self.grid.vert_wall(coord[0], coord[1], room_h)
+
+        # Bottom wall and door
+        for coord in self.hor_wall_positions:
+            self.grid.horz_wall(coord[0], coord[1], room_w)
+
+        for pos in self.doorway_positions:
+            self.grid.set(*pos, None)
+
+        # goal allocation
+        # place goal
+        goal = Goal(self.world, 0)
+        self.put_obj(goal, *self.goal_positions[self.env_seed])
+        goal.init_pos, goal.cur_pos = self.goal_positions[self.env_seed]
+
+        # lava allocation
+        lava_spawn_location = random.sample([0, 1], 1)[0]
+        random_lava_positions = random.sample(
+            self.lava_positions[lava_spawn_location], 3
+        )
+        for lava_pos in random_lava_positions:
+            lava = Lava(self.world)
+            self.put_obj(lava, *lava_pos)
+
+        # agent allocation
+        self.place_agent(self.agents[0], pos=self.agent_positions[self.env_seed])
+
+    def reset(
+        self,
+        *,
+        seed: int | None = None,
+        options: dict | None = None,
+    ):
+        ### intentional to not feed seed since the grid and agent are fixed
```

Comment:
A seed is usually only fed at the start of the first episode. If you'd like to randomly generate the grid once and fix it, you can save the configuration of the grid somehow after the second episode.